### PR TITLE
Use more ES6/7 features

### PR DESCRIPTION
**Pull request by yitzchak**
Decrease dependence on lodash by using more ES6/7 features

- [x] `Array.filter`
- [x] `Array.find`
- [x] `Array.forEach`
- [x] `Array.includes`
- [x] `Array.map`
- [x] `Array.reduce`
- [x] `Array.some`
- [x] `Object.assign`
- [x] More use of object destructuring including rest

### CODE REVIEW

File: lib/loggers/default-logger.js [hunk 1]
```diff
@@ -53,7 +52,7 @@ export default class DefaultLogger extends Logger {
   showErrorMarkersInEditor (editor, messages) {
     const filePath = editor.getPath()
     if (filePath) {
-      const m = _.filter(messages, message => {
+      const m = messages.filter(message => {
         return message.filePath && message.range && filePath.includes(message.filePath)
```
Comment (thomasjo) [inline lib/loggers/default-logger.js]: Let's nuke the braces so we can nuke the `return`.

File: lib/logger.js [hunk 2]
```diff
@@ -46,7 +46,7 @@ export default class Logger {
     const loggingLevel = atom.config.get('latex.loggingLevel')
     const showBuildWarning = loggingLevel !== 'error'
     const showBuildInfo = loggingLevel === 'info'
-    const filteredMessages = _.filter(this.messages, (message) => {
+    const filteredMessages = this.messages.filter(message => {
       return message.type === 'error' || (showBuildWarning && message.type === 'warning') || (showBuildInfo && message.type === 'info')
```
Comment (thomasjo) [inline lib/logger.js]: Remove braces + return

File: lib/loggers/default-logger.js [hunk 3]
```diff
@@ -53,9 +52,8 @@ export default class DefaultLogger extends Logger {
   showErrorMarkersInEditor (editor, messages) {
     const filePath = editor.getPath()
     if (filePath) {
-      const m = _.filter(messages, message => {
-        return message.filePath && message.range && filePath.includes(message.filePath)
-      })
+      const m = messages.filter(message =>
```
Comment (thomasjo) [inline lib/loggers/default-logger.js]: Rename `m` as `marker`?

Comment (yitzchak) [inline lib/loggers/default-logger.js]: Sounds good.

Comment (thomasjo) [inline lib/loggers/default-logger.js]: Wow, less than a minute response time — you're a 🤖, hehe!

Comment (yitzchak) [inline lib/loggers/default-logger.js]: Ha! It's because I am working on that branch right now. More commits soon!!!

**Comment by yitzchak** [comment]
What about changing `heredoc` from

```javascript
heredoc (input) {
  if (input === null) { return null }

  const lines = _.dropWhile(input.split(/\r\n|\n|\r/), line => line.length === 0)
  const indentLength = _.takeWhile(lines[0], char => char === ' ').length
  const truncatedLines = lines.map(line => line.slice(indentLength))

  return truncatedLines.join('\n')
}
```

with something like

```javascript
heredoc (input) {
  if (input === null) { return null }

  const match = input.match(/^ +/m)
  if (match) {
    input = input.replace(new RegExp(`^${match[0]}`, 'mg'), '')
  }

  return input.trim()
}
```

Thereby removing lodash dependence there.

**Pull request merged**